### ISSUE TITLE
Azure Functions host - Lease lock http 409

### ISSUE THREAD

**Issue by hsubramaniam** [Azure/azure-functions-host#4853]
#### Question

I have a Azure function triggered by Azure Service Bus topic. Occasionally I see a lot of http 409 errors in Application Insights when the functions host tries to acquire a lock i believe and persist that lock in Azure blob storage.

url - https://<functionstoragename>.blob.core.windows.net:443/azure-webjobs-hosts/locks/functionappname/host?comp=lease

Maxconcurrentcalls is set at 16

What is causing this and where is this documented?

**Comment by hsubramaniam** [comment]
> Are you seeing these in the Dependencies in app insights?
> 
> The 409s for lease ownership are expected and not actually errors. When a host starts, it attempts to acquire a lock -- if another host has it, it gets a 409 back. App Insights treats anything as 400+ as an error, so it flags these. If you can share exactly where you're seeing it, we can see if there's any way we can minimize it.

@brettsam Yes, I am seeing it on Dependencies in app insights.  Its good to know these are not errors that affect function execution. However, too many such exceptions thrown, especially in a high volume use cases, could quite possibly decrease the overall throughput. And decrease in throughput could mean more time needed to drain the queues and a fraction of increase in costs. Would be great if this is kept this to a minimum

![image](https://user-images.githubusercontent.com/53438549/63911994-33e83480-ca24-11e9-9094-b28d995e60ea.png)

**Comment by brettsam** [comment]
I do get a lot of questions about this. We currently just use the Storage SDK to check for locks, which results in these 409s behind the scenes. But we could:

1. Check if the lease is available.
2. If so, try to acquire it. (it may be a race and you still get a 409)

This would result in more HTTP requests at startup, but it would reduce a lot of steady-state 409s.
